core/src/bms/player/beatoraja/select/BarRenderer.java

```diff
@@ -407,6 +407,93 @@ public Map<String, Object> getFilter() {
 		public void setFilter(Map<String, Object> filter) {
 			this.filter = filter;
 		}
+
+		public Boolean filterSong(ScoreData scoreData) {
+			Set<String> filterKey = this.getFilter().keySet();
+			for (String key : filterKey) {
+				String getterMethodName = "get" + key.substring(0, 1).toUpperCase()
+						+ key.substring(1);
+				try {
+					if (this.getFilter().get(key) instanceof Integer) {
+						Integer value = (Integer)this.getFilter().get(key);
+						if (scoreData == null) {
+							if (0 != value) {
+								return false;
+							}
+						} else {
+							Method getterMethod = ScoreData.class.getMethod(getterMethodName);
+							Object propertyValue = getterMethod.invoke(scoreData);
+							if (!propertyValue.equals(value)) {
+								return false;
+							}
+						}
+						return true;
+					}
+					Object valueArr[] = ((String)this.getFilter().get(key)).split("&&");
+					for (Object value : valueArr)
+					{
+						value = ((String)value).replaceAll("\\s","");
+						if (scoreData == null) {
+							if (value instanceof String) {
+								String stringValue = (String) value;
+								if (!stringValue.isEmpty() && !stringValue.substring(0, 1).equals("<")) {
+									return false;
+								}
+							}
+						} else {
+							Method getterMethod = ScoreData.class.getMethod(getterMethodName);
+							Object propertyValue = getterMethod.invoke(scoreData);
+							if (propertyValue instanceof Integer) {
+								String valueString = (String)value;
+								Integer propertyValueInt = (Integer)propertyValue;
+								Integer filterValueInt;
+								if (valueString.substring(1,2).equals("=")){
+									filterValueInt = Integer.parseInt(valueString.substring(2,valueString.length()));
```

Comment:
You don't need to include the length parameter if you are taking a substring to the end of the string